src/BotPlutusInterface/Contract.hs

```diff
@@ -446,3 +440,92 @@ currentTime contractEnv =
   currentSlot @w contractEnv
     >>= slotToPOSIXTime @w
     >>= either (error . show) return
+
+-- | Check if collateral in contract environment, if not - create and set to environment
+handleCollateral ::
+  forall (w :: Type) (effs :: [Type -> Type]).
+  Member (PABEffect w) effs =>
+  ContractEnvironment w ->
+  Eff effs (Either Text ())
+handleCollateral cEnv = do
+  result <- (fmap swapEither . runEitherT) $
+    do
+      collateralNotInMem <-
+        newEitherT $
+          swapEither . maybeToRight "Collateral UTxO not found in contract env."
+            <$> getInMemCollateral @w
+
+      helperLog collateralNotInMem
+
+      collateralNotInWallet <- newEitherT $ swapEither <$> findCollateralAtOwnPKH cEnv
+
+      helperLog
+        ("Collateral UTxO not found or failed to be found in wallet: " <> pretty collateralNotInWallet)
+
+      helperLog "Creating collateral UTxO."
+      --
+      notCreatedCollateral <- newEitherT $ swapEither <$> makeCollateral @w cEnv
+
+      helperLog
+        ("Failed to create collateral UTxO: " <> pretty notCreatedCollateral)
+
+      return ("Failed to create collateral UTxO: " <> notCreatedCollateral)
+
+  case result of
+    Right collteralUtxo ->
+      setInMemCollateral @w collteralUtxo
+        >> Right <$> printBpiLog @w Debug "successfully set the collateral utxo in env."
+    Left err -> pure $ Left $ "Failed to make collateral: " <> err
+  where
+    --
+    helperLog :: PP.Doc () -> ExceptT CollateralUtxo (Eff effs) ()
+    helperLog msg = newEitherT $ Right <$> printBpiLog @w Notice msg
```

Comment:
Before merging it to main I think log level should be changed to `Debug`.
I've used `Notice` just to make logs less noisy during development.

Reply:
Fixed. Thanks! for pointing it out.